### DISCUSSION TITLE
How to execute a custom function as admin instead of the logged in user who triggerd the workflow?

### DISCUSSION THREAD

**Discussion by LaurensCleyman** [YetiForceCompany/YetiForceCRM#16933] [How to]
Hi, 

I've created a a workflow which is triggered **Every time a record is saved (eg. creating, modifying)**, 
the workflow executes a custom function to update some fields, just like the one below.

Currently the function is executed as the user who created or modified the record that triggered this workflow.
How can I execute this function as another user? For example the admin user?

```
$recordModel = \Vtiger_Record_Model::getCleanInstance($rawRecordModel->getModuleName());
$recordModel->setData($rawRecordModel->getData());
$recordModel->ext = $rawRecordModel->ext;
$recordModel->isNew = false;

// my operations

$recordModel->setHandlerExceptions(['disableHandlerClasses' => ['Vtiger_Workflow_Handler']]);
$recordModel->save();
```

I know it 's possible to execute it as an adminitrator by scheduling the workflow, but the function has to be executed directly after creation/modifing a record. 

Any suggestions? :-)

**Comment by rskrzypczak** [comment]
Why do you need an admin here?
I suppose in terms of modifying fields or searching for records. 
You can omit permissions in any of these options.
https://github.com/YetiForceCompany/YetiForceCRM/blob/b8709bc012b91ca949ecb865b32c2c7b550e0b72/modules/com_vtiger_workflow/tasks/VTAutoAssign.php#L42
https://github.com/YetiForceCompany/YetiForceCRM/blob/b8709bc012b91ca949ecb865b32c2c7b550e0b72/modules/Vtiger/uitypes/Token.php#L80
